packages/expo-doctor/src/checks/ExpoConfigCommonIssueCheck.ts

```diff
@@ -0,0 +1,55 @@
+import JsonFile from '@expo/json-file';
+import resolveFrom from 'resolve-from';
+
+import { DoctorCheck, DoctorCheckParams, DoctorCheckResult } from './checks.types';
+
+export class ExpoConfigCommonIssueCheck implements DoctorCheck {
+  description = 'Check Expo config for common issues';
+
+  sdkVersionRange = '*';
+
+  async runAsync({ projectRoot, exp }: DoctorCheckParams): Promise<DoctorCheckResult> {
+    const issues: string[] = [];
+    let advice;
+
+    // compare SDK version in package.json with installed expo package version
+    // If these don't match, it almost certainly means the user specified expo.sdkVersion in their app.json/ app.config.js
+    const expoSDKVersionFromPackageJson = getExpoSDKVersionFromPackage(projectRoot);
+
+    if (expoSDKVersionFromPackageJson !== exp.sdkVersion) {
+      issues.push(
+        "It appears that expo.sdkVersion is defined in your app.json/ app.config.js. This can cause 'expo install' to install dependency versions for the wrong SDK. SDK version is determined by the version of the expo package installed in your project."
+      );
+      advice = 'Remove expo.sdkVersion from your app.json/ app.config.js.';
+    }
+
+    return {
+      isSuccessful: issues.length === 0,
+      issues,
+      advice,
+    };
+  }
+}
+
+/**
+ * Resolve the Expo SDK Version either from the input Expo config or from the installed
+ * version of the `expo` package.
+ * Adapted from https://github.com/expo/expo/blob/main/packages/%40expo/config/src/getExpoSDKVersion.ts
+ */
+function getExpoSDKVersionFromPackage(projectRoot: string): string | undefined {
```

Comment:
Is this something we can ask `@expo/config`? If so, I think that might be a better bet. The main reason for that is that we probably want `@expo/config` to handle all configuration loading, including the app manifest.

When we change something in a future version of `@expo/config`, we also need to make that change here. That is, unless we use the installed `@expo/config` package from the project and just use that here.

I do the same for `expo-github-action`, where I pull the configuration file, required to retrieve basic information for the update / preview sub-action. ([see this](https://github.com/expo/expo-github-action/blob/main/src/expo.ts#L146-L160))